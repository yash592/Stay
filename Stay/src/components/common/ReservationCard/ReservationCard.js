import React, { Component } from "react";
import { View, Text, TouchableOpacity, Image } from "react-native";

// ReservationCard reusable pure component which can be used across the app

const ReservationCard = props => {
  // console.log(props);
  return (
    <TouchableOpacity style={styles.container}>
      {props.children}
      <View style={styles.nameBox}>
        <Text style={styles.nameText}>{props.name}</Text>
      </View>
      <View style={styles.arrivalDateBox}>
        <Image
          source={require("../../assets/images/arrival.png")}
          style={{ width: 28, height: 28, opacity: 0.5 }}
        />
        <Text numberOfLines={1} style={styles.text}>
          {props.arrivalDate}
        </Text>
      </View>
      <View style={styles.departureDateBox}>
        <Image
          source={require("../../assets/images/departures.png")}
          style={{ width: 28, height: 28, opacity: 0.5 }}
        />
        <Text numberOfLines={1} style={styles.text}>
          {props.departureDate}
        </Text>
      </View>
      <View style={styles.locationBox}>
        <Image
          source={require("../../assets/images/location-pin.png")}
          style={{ width: 28, height: 28, opacity: 0.5 }}
        />
        <Text numberOfLines={2} style={styles.text}>
          {props.hotelName}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = {
  container: {
    flex: 1,
    marginTop: 30,
    width: "95%",
    height: 150,
    borderRadius: 10,
    shadowColor: "black",
    shadowOffset: { width: 10, height: 10 },
    shadowOpacity: 1,
    shadowRadius: 1,
    elevation: 5,
    backgroundColor: "white"
  },
  imageBox: {
    width: 100
  },
  image: {
    width: "100%",
    height: 100
  },
  nameBox: {
    position: "absolute",
    left: "5%",
    top: "10%"
  },
  arrivalDateBox: {
    position: "absolute",
    top: "10%",
    right: "5%"
  },
  departureDateBox: {
    position: "absolute",
    bottom: "10%",
    right: "5%"
  },
  locationBox: {
    position: "absolute",
    width: "50%",

    flexWrap: "wrap",
    bottom: "10%",
    left: "5%"
  },
  nameText: {
    fontSize: 30,
    color: "black",
    fontWeight: "normal",
    fontFamily: "sans-serif-condensed"
  },
  text: {
    fontSize: 18,
    fontFamily: "sans-serif-condensed"
  }
};

export { ReservationCard };
