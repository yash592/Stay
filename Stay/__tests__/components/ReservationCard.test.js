import "react-native";
import renderer from "react-test-renderer";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";

configure({ adapter: new Adapter() });
import { ReservationCard } from "../../src/components/common/ReservationCard";
import { shallow } from "enzyme";

describe("ReservationCard component renders correctly", () => {
  it("Shallow rendering", () => {
    const component = shallow(<ReservationCard />);
    expect(component).toMatchSnapshot();
  });
});
