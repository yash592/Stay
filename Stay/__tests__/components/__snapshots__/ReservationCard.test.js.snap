// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationCard component renders correctly Shallow rendering 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 5,
      "flex": 1,
      "height": 150,
      "marginTop": 30,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 10,
        "width": 10,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "left": "5%",
        "position": "absolute",
        "top": "10%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "sans-serif-condensed",
          "fontSize": 30,
          "fontWeight": "normal",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "position": "absolute",
        "right": "5%",
        "top": "10%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/components/assets/images/arrival.png",
        }
      }
      style={
        Object {
          "height": 28,
          "opacity": 0.5,
          "width": 28,
        }
      }
    />
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontFamily": "sans-serif-condensed",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "bottom": "10%",
        "position": "absolute",
        "right": "5%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/components/assets/images/departures.png",
        }
      }
      style={
        Object {
          "height": 28,
          "opacity": 0.5,
          "width": 28,
        }
      }
    />
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontFamily": "sans-serif-condensed",
          "fontSize": 18,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "bottom": "10%",
        "flexWrap": "wrap",
        "left": "5%",
        "position": "absolute",
        "width": "50%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/components/assets/images/location-pin.png",
        }
      }
      style={
        Object {
          "height": 28,
          "opacity": 0.5,
          "width": 28,
        }
      }
    />
    <Text
      numberOfLines={2}
      style={
        Object {
          "fontFamily": "sans-serif-condensed",
          "fontSize": 18,
        }
      }
    />
  </View>
</TouchableOpacity>
`;
